src/back-end/lib/resources/affiliation.ts

```diff
@@ -59,47 +60,56 @@ const resource: Resource = {
   create(connection) {
     return {
       async parseRequestBody(request) {
-        const body = request.body.tag === 'json' ? request.body.value : {};
+        const body: unknown = request.body.tag === 'json' ? request.body.value : {};
         return {
-          user: getString(body, 'user'),
+          userEmail: getString(body, 'userEmail'),
           organization: getString(body, 'organization'),
           membershipType: getString(body, 'membershipType')
         };
       },
       async validateRequestBody(request) {
-        const { user, organization, membershipType } = request.body;
-        const validatedUser = await validateUserId(connection, user);
+        const { userEmail, organization, membershipType } = request.body;
+        // Attempt to fetch a valid user for the given email
+        const validatedUser = await db.readOneUserByEmail(connection, userEmail);
         const validatedOrganization = await validateOrganizationId(connection, organization);
         const validatedMembershipType = affiliationValidation.validateMembershipType(membershipType);
         if (allValid([validatedUser, validatedOrganization, validatedMembershipType])) {
-          // Get the most recent affiliation for this user and organization
-          const dbResult = await db.readOneAffiliation(connection, user, organization);
+          const user = validatedUser.value as User | null;
```

Comment:
You can change this to `getValidValue(validatedUser, null)` to avoid the type cast.